bgpd/bgp_route.c

```diff
@@ -2465,33 +2466,37 @@ bool subgroup_announce_check(struct bgp_dest *dest, struct bgp_path_info *pi,
 	 * we do not announce LL address as `::`.
 	 */
 	if (NEXTHOP_IS_V6) {
-		attr->mp_nexthop_len = BGP_ATTR_NHLEN_IPV6_GLOBAL;
-		if ((CHECK_FLAG(peer->af_flags[afi][safi],
-				PEER_FLAG_NEXTHOP_LOCAL_UNCHANGED) &&
-		     IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_local)) ||
-		    (!reflect && !transparent &&
-		     IN6_IS_ADDR_LINKLOCAL(&peer->nexthop.v6_local) &&
-		     peer->shared_network &&
-		     ((from == bgp->peer_self && peer->sort == BGP_PEER_EBGP) ||
-		      (from == bgp->peer_self && peer->sort != BGP_PEER_EBGP) ||
-		      (from != bgp->peer_self &&
-		       IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_local) &&
-		       peer->sort == BGP_PEER_EBGP)))) {
+		if (CHECK_FLAG(peer->af_flags[afi][safi], PEER_FLAG_NEXTHOP_LOCAL_UNCHANGED)) {
```

Comment:
So, if we don't have `nexthop-local unchanged` and the peer is a route-server-client we don't send LL in any case?

Reply:
Yes, we don't and that is correct. See bgp_route_server_client topotest

See the individual commits. I do not change that.